pkg/cli/admin/upgrade/status/mco/layered_node_state.go

```diff
@@ -85,7 +85,10 @@ func (l *LayeredNodeState) isImageAnnotationEqualToPool(anno string, mcp *mcfgv1
 		if lps.GetOSImage() == val {
 			return true
 		}
-		l.ReasonOfUnavailability = fmt.Sprintf("the node annotation %s is %s, while the pool is trying to reconcile OS image %s", anno, lps.GetOSImage(), val)
+		// According to https://github.com/openshift/machine-config-operator/pull/4510#issuecomment-2271461847
+		// ExperimentalNewestLayeredImageEquivalentConfigAnnotationKey is not used any more and this case should never happen.
+		klog.V(5).Infof("The node annotation %s has value %s different from the OS image %s", anno, val, lps.GetOSImage())
+		l.ReasonOfUnavailability = fmt.Sprintf("the node has an unexpected annotation %s=%s", ExperimentalNewestLayeredImageEquivalentConfigAnnotationKey, lps.GetOSImage())
```

Comment:
I am not sure whether we should hardcode the `ExperimentalNewestLayeredImageEquivalentConfigAnnotationKey` const instead of the variable `anno` here or have the reason to know the context of the key's present/future existence. IMO we could just use the log message and use it slightly modified, as it adheres to the local code logic and not the context of usage of the `ExperimentalNewestLayeredImageEquivalentConfigAnnotationKey` annotation in OpenShift or the logic of the `HasOSImage` method.

I would maybe prefer something like this. WDYT?
```go
l.ReasonOfUnavailability = fmt.Sprintf("the node OS image %s differs from its machine config pool OS image %s", val, lps.GetOSImage())
```

Reply:
`Node has an unexpected annotation %s=%s` sounds more accessible to for a standard OCP user than `node OS image differs from its machine config pool OS image` though? I would not expect most users to have knowledge of how a node OS images and how they are managed in MCP -> Node hierarchy (but correct output for "should never happen" cases is mostly bikeshedding so :shrug: )

Reply:
Let's not waste time discussing this, as it should not happen. `Node has an unexpected annotation %s=%s` has a majority vote so let's keep it.

Reply:
Nice to know what you think. Thanks.